packages/x-components/src/components/suggestions/base-suggestions.vue

```diff
@@ -106,7 +122,48 @@
      * @internal
      */
     protected get suggestionsToRender(): Suggestion[] {
-      return this.suggestions.slice(0, this.maxItemsToRender);
+      const suggestions = this.suggestions.slice(0, this.maxItemsToRender);
```

Comment:
I was working in a better approach, im not sure 100% on it, i want to check it again, but here is the suggestion:

```
    protected get suggestionsToRender(): Suggestion[] {
      const suggestions = this.suggestions.slice(0, this.maxItemsToRender);

      return this.showFacets
        ? suggestions.reduce<Suggestion[]>((acc, suggestion) => {
            if (suggestion.facets && suggestion.facets.length > 0) {
              acc = [...acc, ...this.generateSuggestionsFromFacets(suggestion)];
            } else {
              acc = [...acc, { ...suggestion, facets: [] }];
            }
            return acc;
          }, [])
        : suggestions.map(suggestion => ({ ...suggestion, facets: [] }));
```

```
 protected generateSuggestionsFromFacets(suggestion: Suggestion): Suggestion[] {
      const suggestionsWithFacets = suggestion.facets.reduce<Suggestion[]>((suggestions, facet) => {
        facet.filters.forEach(filter => {
          suggestions.push({
            ...suggestion,
            facets: [
              {
                ...facet,
                filters: [filter]
              }
            ]
          });
        });
        return suggestions;
      }, []);
      if (this.showQuery) {
        suggestionsWithFacets.unshift({ ...suggestion, facets: [] });
      }
      return suggestionsWithFacets;
    }
    ```
    
   I am still trying to find another approach

Reply:
I like this approach better, indeed. I might extract the ternary statements into its own functions for readability.